src/Cli/dotnet/commands/dotnet-msbuild/MSBuildLogger.cs

```diff
@@ -138,6 +141,21 @@ internal static void FormatAndSend(ITelemetry telemetry, TelemetryEventArgs args
                         toBeHashed: Array.Empty<string>(),
                         toBeMeasured: new[] { "FileLoggersCount" });
                     break;
+                case BuildcheckAcquisitionFailureEventName:
+                    TrackEvent(telemetry, $"msbuild/{BuildcheckAcquisitionFailureEventName}", args.Properties,
+                        toBeHashed: new[] { "AssemblyName", "ExceptionType", "ExceptionMessage" },
+                        toBeMeasured: Array.Empty<string>());
+                    break;
+                case BuildcheckRunEventName:
+                    TrackEvent(telemetry, $"msbuild/{BuildcheckRunEventName}", args.Properties,
+                        toBeHashed: Array.Empty<string>(),
+                        toBeMeasured: new[] { "TotalRuntimeInMilliseconds" });
+                    break;
+                case BuildcheckRuleStatsEventName:
+                    TrackEvent(telemetry, $"msbuild/{BuildcheckRuleStatsEventName}", args.Properties,
+                        toBeHashed: new[] { "RuleId", "CheckFriendlyName" },
```

Comment:
@dsplaisted FYI - 

* the RuleId is the MSBuild diagnostic id, like "BC0103" or similar. We're hashing it because it might leak product names, etc
* the CheckFriendlyName is a more descriptive name like "MSBuild.DoubleWritesCheck" or similar, but since it is free-form it could contain sensitive information so we're hashing it